src/main/java/org/checkstyle/autofix/parser/ConfigurationLoader.java

```diff
@@ -37,38 +36,52 @@ private ConfigurationLoader() {
         // utility class
     }
 
-    public static CheckConfiguration mapConfiguration(Configuration config) {
-        final Map<String, String> properties = new HashMap<>();
-        final String[] propertyNames = config.getPropertyNames();
-        for (String propertyName : propertyNames) {
-            try {
-                final String value = config.getProperty(propertyName);
-                properties.put(propertyName, value);
+    public static Map<CheckstyleCheck, CheckConfiguration> mapConfiguration(Configuration config) {
+        final Map<CheckstyleCheck, CheckConfiguration> result = new HashMap<>();
+        final Map<String, String> inherited = getProperties(config);
+
+        final CheckstyleCheck module = getModule(config.getName());
+        if (module != null) {
+            result.put(module, new CheckConfiguration(module,
+                    new HashMap<>(), getProperties(config)));
+        }
 
+        for (Configuration child : config.getChildren()) {
+            mapConfiguration(child).forEach((childModule, childConfig) -> {
+                inherited.forEach(childConfig::setGlobalProperty);
+                result.put(childModule, childConfig);
+            });
+        }
+
+        return result;
+    }
+
+    private static Map<String, String> getProperties(Configuration config) {
+        final Map<String, String> props = new HashMap<>();
+        for (String prop : config.getPropertyNames()) {
+            try {
+                props.put(prop, config.getProperty(prop));
             }
             catch (CheckstyleException exception) {
-                throw new IllegalStateException("Error getting property " + propertyName,
-                        exception);
+                throw new IllegalStateException("Failed to get property: " + prop, exception);
             }
         }
+        return props;
+    }
 
-        final Configuration[] checkstyleChildren = config.getChildren();
-        final CheckConfiguration[] simpleChildren =
-                new CheckConfiguration[checkstyleChildren.length];
-        for (int index = 0; index < checkstyleChildren.length; index++) {
-            simpleChildren[index] = mapConfiguration(checkstyleChildren[index]);
+    private static CheckstyleCheck getModule(String name) {
+        CheckstyleCheck module;
+        try {
+            module = CheckstyleCheck.valueOf(name.toUpperCase(Locale.ROOT));
```

Comment:
we should use "fromSource". Please update "fromSource" to handle both packaged prefixed names and simple name. Given we use ids in the CheckstyleCheck, should be enough to iterate over ids and check if it contains substring, which is "name". This should also allow to use proper enum names: "UPPER_ELL" instead of "UPPERELL"